graphs/util/db.py

```diff
@@ -1608,6 +1608,298 @@ def find_all_graphs_containing_nodes(uid, search_type, search_word, view_type, d
 
 	return graph_dict.values()
 
+
+def uploadGPMLFile(username, graphJSON, title):
+	'''
+		Uploads GPML file to GraphSpace via /upload.
+		@param username: Owner of graph
+		@param graphJSON: JSON of graph
+		@param title: Title of graph
+	'''
+
+	try:
+		# Create JSON stucture for GraphSpace recognized JSON
+		parseJson = {"graph": {"edges": [], "nodes": []}, "metadata": {}}
+
+		# Use xml parser to parse the GPML file
+		import xml.etree.ElementTree as ET
+		tree = ET.fromstring(graphJSON)
+
+		# The current GPML format
+		GPML = '{http://pathvisio.org/GPML/2013a}'
+
+		# Keep the overall count of nodes and create unique ids for datanodes, shape, labels.
+		# IMPORTANT!!!
+		count = 0
+
+		# Parsing the GPML file for Shape elements
+		shapes = tree.findall(GPML+'Shape')
+		# Create a list of details of all the shape
+		shape_details = []
+		for shape in shapes:
+			tempNode = {}
+			tempNode.update(shape.attrib)
+			for more_details in shape:
+				tempNode.update(more_details.attrib)
+			shape_details.append(tempNode)
+
+		# Add shape elemets to the JSON structure parseJson
+		for node in shape_details:
+			tempNode = {'data' : {}}
+
+			# Give an id to the shape element.
+			tempNode['data']['id'] = count
+			if not ('CenterX' in node and 'CenterY' in node):
+				return {"Error": "GPML file must contain X and Y coordinates of all the shape elements!"}
+			tempNode['data']['x'] = node['CenterX']
+			tempNode['data']['y'] = node['CenterY']
+
+			# Shape element may or may not contain a TextLabel field
+			if 'TextLabel' in node:
+				tempNode['data']['content'] = node['TextLabel']
+
+			# Deal with the ShapeType field of the shape element,
+			# improve this, make it more general, too many hacks right now.
+			if 'ShapeType' in node:
+				if node['ShapeType'].lower() == 'brace':
+					tempNode['data']['shape'] = 'rectangle'
```

Comment:
`if node['ShapeType'].lower() == 'brace'`

Can you define a constant variable in graphspace graph class for 'brace' and 'ShapeType' ? Also do the same for other constant strings as well.